loading-code/typescript-default-extenstion/src/systemjs-hooks/fetch.ts

```diff
@@ -0,0 +1,14 @@
+//@ts-ignore
+System.shouldFetch = () => true
+
+const endsWith = (str: string, suffix: string) => str.indexOf(suffix, str.length - suffix.length) !== -1
+
+const origFetch = System.constructor.prototype.fetch
+System.constructor.prototype.fetch = function (url: string) {
```

Comment:
I don't think the fetch hook is the right one to use here. Fetch + eval is worse for performance than a simple script tag append. The `resolve()` hook would be more appropriate.

See [this codesandbox](https://codesandbox.io/s/mystifying-wave-unfh2) for how this can be implemented with the resolve hook.

Reply:
My first approach was to do it using the resolve hook as well. All was fine until you have an importmap where an entry has no file ending. The resolve hook seems to run before the importmap is resolved. I'm on mobile now, but this was the reason that I used the fetch hook instead. 

I might be wrong here, since I'm quite new to SystemJS, and I'm probably missing something basic. 

Will try to create an example of what I mean during the weekend.

Reply:
@joeldenning I was wrong - in my last attempt using `resolve`, I wrote my code something like this: 

```
const endsWithFileExtension = /\/?\.[a-zA-Z]{2,}$/;
        const originalResolve = System.constructor.prototype.resolve;
        System.constructor.prototype.resolve = function(url, ...args) {
          url = endsWithFileExtension.test(url) ? url : url + ".js";
          return originalResolve.apply(this, url, ...args);
        };
```

When writing like this, the importmap wasn't applied until afterwards, which of course didn't work. Your code works great. Thanks!